.github/workflows/publish.yaml

```diff
@@ -30,21 +30,25 @@ jobs:
     - name: Install Skopeo
       run: sudo snap install --devmode --channel edge skopeo
 
-    - name: Upload ROCK to ghcr.io with latest
-      id: latest
-      if: github.ref_type == 'branch'
+    - name: Upload ROCK to ghcr.io
+      id: upload
       run: |
-        sudo skopeo --insecure-policy copy oci-archive:$(realpath ./"${{ inputs.rock }}") docker://ghcr.io/canonical/identity-platform-login-ui:"${{ github.sha }}" --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" 
-        sudo skopeo --insecure-policy copy oci-archive:$(realpath ./"${{ inputs.rock }}") docker://ghcr.io/canonical/identity-platform-login-ui:latest --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}"
-        echo "image=ghcr.io/canonical/identity-platform-login-ui:${{ github.sha }}" >> "$GITHUB_ENV"
-    - name: Upload ROCK to ghcr.io with stable
-      id: stable
-      if: github.ref_type == 'tag'
-      run: |
-        sudo skopeo --insecure-policy copy oci-archive:$(realpath ./"${{ inputs.rock }}") docker://ghcr.io/canonical/identity-platform-login-ui:"${{ github.ref_name }}" --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" 
-        sudo skopeo --insecure-policy copy oci-archive:$(realpath ./"${{ inputs.rock }}") docker://ghcr.io/canonical/identity-platform-login-ui:stable --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}"
-        echo "image=ghcr.io/canonical/identity-platform-login-ui:${{ github.ref_name }}" >> "$GITHUB_ENV"
+        if [ "${{ github.ref_type }}" == "branch" ]; then
+          versions=(latest "${{github.sha}}")
+        fi
+
+        if [ "${{ github.ref_type }}" == "tag" ]; then
+          versions=(stable "${{ github.ref_name }}")
+        fi
+
+        for version in "${versions[@]}"; do
+          sudo skopeo --insecure-copy \
+            --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" \
+            oci-archive:$(realpath ./"${{ inputs.rock }}") \
+            docker://ghcr.io/canonical/identity-platform-login-ui:"${version}"
+          echo "image=ghcr.io/canonical/identity-platform-login-ui:${version}" >> "$GITHUB_ENV"
+        done
```

Comment:
Just to be sure: this does not alter the functionality of the upload step, it just merges the 2.

Do you think that having the `if` in shell is clearer than using `if` of the workflow syntax? I think that the `if` in the workflow provides more readability (different name depending on the channel), but I don't have a strong preference either way (cc @shipperizer )

Reply:
i'd prefer the if in the workflow as well tbh, even if it causes repetition